llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -4727,7 +4732,33 @@ SDValue AMDGPUTargetLowering::performSelectCombine(SDNode *N,
   SDValue True = N->getOperand(1);
   SDValue False = N->getOperand(2);
 
-  if (Cond.hasOneUse()) { // TODO: Look for multiple select uses.
+  int ShouldSwap = 0;
+  for (auto it = Cond->use_begin(); it != Cond->use_end(); it++) {
+    auto User = it->getUser();
+
+    if (User->getOpcode() != ISD::SELECT) {
+      ShouldSwap = 0;
+      break;
+    }
+
+    auto Op1 = User->getOperand(1);
+    auto Op2 = User->getOperand(2);
+
+    // if the operand is defined by fneg or fabs it means the instruction
+    // will have source modifiers and therefore can't be shrinked to vop2
+    if (isFnegOrFabs(Op1) || isFnegOrFabs(Op2))
+      continue;
+
+    bool IsOp1Divergent = Op1->isDivergent();
+    bool IsOp2Divergent = Op2->isDivergent();
+
+    if (!IsOp1Divergent && IsOp2Divergent)
+      ShouldSwap++;
+    else if (IsOp1Divergent && !IsOp2Divergent)
+      ShouldSwap--;
+  }
+
+  if (Cond->hasOneUse() || ShouldSwap > 0) {
```

Comment:
The `hasFminFmaxLegacy` code below should almost certainly stay inside an `if (Cond->hasOneUse())` check. It is not related to your `ShouldSwap` logic.